File: 1299. Replace Elements with Greatest Element on Right Side/1299.go
package leetcode

func replaceElements(arr []int) []int {
	maxnum := -1
	for i:=len(arr)-1; i>-1; i--{
		tmp := arr[i]
		arr[i] = maxnum
		if tmp>maxnum{
			maxnum = tmp
		}
	}
	return arr
}

File: 1299. Replace Elements with Greatest Element on Right Side/_1299.py
class Solution:
    def replaceElements(self, arr) :
        maxnum = -1
        for i in range(len(arr)-1, -1, -1):
            tmp = arr[i]
            arr[i] = maxnum
            maxnum = max(tmp, maxnum)
        return arr

    def replaceElements1(self, A, mx = -1):
        for i in range(len(A) - 1, -1, -1):
            A[i], mx = mx, max(mx, A[i])
        return A